src/pushsource/_impl/backend/registry_source.py

```diff
@@ -0,0 +1,171 @@
+import logging
+
+
+try:
+    from urllib.parse import urlparse
+except ImportError: # pragma: no cover
+    from urlparse import urlparse
+
+from ..source import Source
+from ..model import (
+    ContainerImagePushItem,
+    SourceContainerImagePushItem,
+    ContainerImagePullInfo,
+    ContainerImageTagPullSpec,
+    ContainerImageDigestPullSpec,
+)
+from ..utils.containers import (
+    get_manifest,
+    inspect,
+    MT_S2_V2,
+    MT_S2_V1,
+    MT_S2_V1_SIGNED,
+    MT_S2_LIST,
+)
+
+from ..helpers import list_argument
+
+LOG = logging.getLogger("pushsource")
+
+
+class RegistrySource(Source):
+    """Uses URIs of container images as source for push items."""
+
+    def __init__(
+        self,
+        dest=None,
+        registry_uris=None,
+        signing_key=None,
+        docker_url="unix://var/run/docker.sock",
+        executor_container_image=None,
+        docker_timeout=None,
+        docker_verify_tls=None,
+        docker_cert_path=None,
+    ):
+        """Create a new source.
+
+        Parameters:
+            dest (str, list[str])
```

Comment:
The docs don't spell out how "dest" and "destination_tag" on the URIs are supposed to work together, but I'll assume the code as written now is as intended.  What's implemented seems to be that "dest" should contain repos, and "destination_tag" should contain tags, and then the final value of "dest" on each generated item is equal to the repos and tags combined.

I think there are some issues with this:

- It's inconsistent with the handling of 'dest' argument for other sources when dealing with container images.  Here's an example taken from [internal docs](https://docs.engineering.redhat.com/x/60QiDQ): `brew:container_build=ubi8-container-8.5-145&dest=ubi/ubi8:latest,ubi/ubi8:8.5` .  If this backend has an argument named "dest", I think it should use the repo:tag in a consistent manner as the existing container-supporting backend.
- Due to the way that the repos in 'dest' and the tags in the URIs are combined together, we end up with the behavior that each individual image referenced can be sent to a different *tag*, but all of them have to be delivered to the same set of *repos*.  Does this meet the use-case actually?  From the point of view of the caller, this seems arbitrary (why can I set tag on a per-image basis, but can only set repos for all images at once).

The cleanest design is not obvious.  How do you expect to calculate the dest in practice, when this is called?

I suggest initially being consistent with koji source, i.e. let it accept a "dest" and just blindly apply this onto all push items, but in practice the non-trivial callers would be expected to set the dest themselves on the generated items.